prow/config/config.go

```diff
@@ -1045,21 +1048,21 @@ func mergeJobConfigs(a, b JobConfig) (JobConfig, error) {
 }
 
 func setPresubmitDecorationDefaults(c *Config, ps *Presubmit, repo string) {
-	if ps.Decorate {
+	if c.DecorateAllJobs || ps.Decorate {
```

Comment:
This prevents someone from defining an undecorated job. We need to be able to define all three things:
* Jobs which inherit the default behavior
* Jobs which must be decorated
* Jobs which must be undecorated.

Long term the default behavior should always be decorated. However I would like to see a migration period:
1) AKA start out with the ability to opt into a decorate default
2) Later the ability to opt out of decorated default
3) Finally requiring decorated defaults (but people should still be able to declare a job that does not use decoration)

Reply:
Thank you for the detailed review. I have added an `Undecorated` option to allow opting out of default behavior and define undecorated jobs.

Reply:
I don't think it's safe to change a current config with no `decorate` set to one that has `decorate: true` so if we follow that path we need a migration period where you have users change their config to set `undecorated` to true